File: PokePlannerWeb.Data/DataStore/Models/LocationEntry.cs
using System.Collections.Generic;

namespace PokePlannerWeb.Data.DataStore.Models
{
    /// <summary>
    /// Represents a location in the data store.
    /// </summary>
    public class LocationEntry : NamedApiResourceEntry
    {
        /// <summary>
        /// Gets the ID of the location.
        /// </summary>
        public int LocationId => Key;

        /// <summary>
        /// Gets or sets the display names of the location.
        /// </summary>
        public List<LocalString> DisplayNames { get; set; }
    }
}

File: PokePlannerWeb/ClientApp/src/models/LocationEntry.ts
import { LocalString } from "./LocalString"

/**
 * Represents a location in the data store.
 */
export class LocationEntry {
    /**
     * The ID of the location.
     */
    locationId: number

    /**
     * The name of the location.
     */
    name: string

    /**
     * The display names of the location.
     */
    displayNames: LocalString[]

    /**
     * Constructor.
     */
    constructor(
        locationId: number,
        name: string,
        displayNames: LocalString[]
    ) {
        this.locationId = locationId
        this.name = name
        this.displayNames = displayNames
    }

    /**
     * Returns a location created from the given entry.
     */
    static from(location: LocationEntry) {
        return new LocationEntry(
            location.locationId,
            location.name,
            location.displayNames
        )
    }

    /**
     * Returns the location's display name in the given locale.
     */
    getDisplayName(locale: string): string | undefined {
        let localName = this.displayNames.find(n => n.language === locale)
        if (localName === undefined) {
            console.warn(
                `Location ${this.locationId} is missing display name in locale '${locale}'`
            )
        }

        return localName?.value
    }
}